### ISSUE TITLE
Feature Request: addon install from zip URL

### ISSUE THREAD

**Issue by funatsufumiya** [chickensoft-games/GodotEnv#63]
(Sorry if duplicated.)

Addons are sometimes published with zip URL with compiled dlls or bins. It would be nice if zip URL is supported on `addons install` :)

For example: https://github.com/godotengine/godot-git-plugin/releases

**Comment by donovani** [comment]
Would also like to see this. Some of the most popular addons are compiled from lower-level languages, making a simple git clone not viable. This would go a long way in user adoption.

For me, this is the case for both:
- [Godot Jolt](https://github.com/godot-jolt/godot-jolt)
- [Debug Draw 3D](https://github.com/DmitriySalnikov/godot_debug_draw_3d)

I'd prefer not to have to automate building every single one of these if possible

**Comment by donovani** [comment]
My guess is that it might be something like this, where a new `source` enum value is defined, which defines how it uses the `url` and `subfolder` properties.

#### Option 1: Directly downloading ZIP
In this case, you just give an arbitrary URL to a zip folder, download it, unzip it to a temporary location, and then use `subfolder` to determine what part of the extracted ZIP to copy.
```json
{
  "addons": {
    "godot-jolt": {
      "url": "https://github.com/godot-jolt/godot-jolt/releases/download/v0.14.0-stable/godot-jolt_v0.14.0-stable.zip",
      "source": "zip",
      "subfolder": "folder/inside/of/zip"
    }
  }
}
```

#### Option 2: Reading the content of a release
In this case, you'd use `url` to find the repo, then `checkout` to find the specific release by name. Since a release on GitHub can have an arbitrary number of build artifacts, you'd probably need another property to state which one to download and unzip - this is what the `artifact` prop is below.
```json
{
  "addons": {
    "godot-jolt": {
      "url": "git@github.com/godot-jolt/godot-jolt.git",
      "source": "release",
      "checkout": "v0.14.0-stable",
      "artifact": " godot-jolt_v0.14.0-stable.zip",
      "subfolder": "folder/inside/of/zip"
    }
  }
}
```

While Option 2 might be more robust and version-friendly, I imagine it opens up a whole can of worms. So to that end, Option 1 might be the simplest, and keep the door open for interacting with releases, should we want to explore something more complex.

**Comment by funatsufumiya** [comment]
@jolexxa 
Add-ons distributed by zip are no different from those distributed by regular git clone. It is just that the size of the dll(s) is so large that it is less compatible with git (for developer), and the compiled result is distributed as a zip with Releases.

I think if simply check if the `content-type` of the URL is a zip, and expand it as a folder after downloading, we can handle it the same way as git.

**Comment by funatsufumiya** [comment]
I think Option 1 is preferable. Because Release function is not a git's feature but a github's feature.
If Option 1 is selected, we can use any URL protocol not dependent to github.

**Comment by donovani** [comment]
Yeah I can agree with that. The benefit of the current implementation is that it's platform agnostic, so that it doesn't matter where it's hosted. But since there's no platform-agnostic implementation of Releases, using the platform-specific APIs would be unavoidable.

I wouldn't be opposed to more integrations with GitHub/GitLab, but it's certainly a lot more complex, and would require reworking how that auth/account info is tracked.

**Comment by donovani** [comment]
> But how do we know if a file has changed on a server and should be re-downloaded? Is there a standardized way to request an MD5 hash of a file from a server via standard HTTPS stuff? 

I'm no expert, but a quick Google seems to suggest that HTTPS & FTP servers don't really have a standardized way of providing a hash. So my guess would be that if we want a hash to track if things change, then it might be unavoidable to download the zip first.

If we know a specific server can provide it, like if GitHub URLs can provide a hash (no idea if they actually do provide this, just using it as an example), then we could perhaps hardcode in some. But it wouldn't be a generalized solution for all URLs.
